src/darker/__main__.py

```diff
@@ -107,7 +107,24 @@ def format_edited_parts(srcs: Iterable[Path], isort: bool) -> None:
                 # 10. A re-formatted Python file which produces an identical AST was
                 #     created successfully - write an updated file
                 logger.info("Writing %s bytes into %s", len(result_str), src)
-                src.write_text(result_str)
+                if print_diff:
+                    from difflib import unified_diff
+
+                    difflines = list(
+                        unified_diff(
+                            open(src).read().splitlines(),
+                            result_str.splitlines(),
```

Comment:
Also, `chosen_lines` already contains the lines of the resulting file after picking the correct reformatted lines.

I'm not sure what the exact behavior there is regarding the final linefeed in the file, could you try and check?

```suggestion
                            chosen_lines,
```